File: sorting/heap_sort.rb
require_relative '../data_structures/heap.rb'

class Heapsort

  def self.sort(list)
    heap = Heap.new
    list.each do |item|
      heap.insert(item)
    end
    result = []
    result.unshift heap.pop while heap.peek
    result
  end

end

File: sorting/heap_sort.cpp
#include <vector>
#include "../data_structures/min_heap.h"

class HeapSort
{
public:
  static std::vector<int> sort(std::vector<int> elements)
  {
    MinHeap heap;
    for(int i=0; i<elements.size(); i++)
    {
      heap.insert(elements[i]);
    }

    std::vector<int> sorted;
    for(int i=0; i<heap.size(); i++)
    {
      sorted.push_back(heap.pop());
    }
    return sorted;
  };
};
